File: src/main/java/in/dragonbra/javasteam/base/ISteamSerializable.java
package in.dragonbra.javasteam.base;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

/**
 * @author lngtr
 * @since 2018-02-21
 */
public interface ISteamSerializable {

    void serialize(OutputStream stream) throws IOException;

    void deserialize(InputStream stream) throws IOException;
}

File: src/main/java/in/dragonbra/javasteam/base/ISteamSerializable.kt
package `in`.dragonbra.javasteam.base

import java.io.IOException
import java.io.InputStream
import java.io.OutputStream

/**
 * @author lngtr
 * @since 2018-02-21
 */
interface ISteamSerializable {
    @Throws(IOException::class)
    fun serialize(stream: OutputStream)

    @Throws(IOException::class)
    fun deserialize(stream: InputStream)
}
